sdk/typescript/src/api/client.ts

```diff
@@ -0,0 +1,814 @@
+		options?: SuiObjectDataOptions;
+	}): Promise<SuiObjectResponse> {
+		if (!input.id || !isValidSuiObjectId(normalizeSuiObjectId(input.id))) {
+			throw new Error('Invalid Sui Object id');
+		}
+		return await this.transport.request({
+			method: 'sui_getObject',
+			params: [input.id, input.options],
+		});
+	}
+
+	async tryGetPastObject(input: {
+		id: ObjectId;
+		version: number;
+		options?: SuiObjectDataOptions;
+	}): Promise<ObjectRead> {
+		return await this.transport.request({
+			method: 'sui_tryGetPastObject',
+			params: [input.id, input.version, input.options],
+		});
+	}
+
+	/**
+	 * Batch get details about a list of objects. If any of the object ids are duplicates the call will fail
+	 */
+	async multiGetObjects(input: {
+		ids: ObjectId[];
+		options?: SuiObjectDataOptions;
+	}): Promise<SuiObjectResponse[]> {
+		input.ids.forEach((id) => {
+			if (!id || !isValidSuiObjectId(normalizeSuiObjectId(id))) {
+				throw new Error(`Invalid Sui Object id ${id}`);
+			}
+		});
+		const hasDuplicates = input.ids.length !== new Set(input.ids).size;
+		if (hasDuplicates) {
+			throw new Error(`Duplicate object ids in batch call ${input.ids}`);
+		}
+
+		return await this.transport.request({
+			method: 'sui_multiGetObjects',
+			params: [input.ids, input.options],
+		});
+	}
+
+	/**
+	 * Get transaction blocks for a given query criteria
+	 */
+	async queryTransactionBlocks(
+		input: SuiTransactionBlockResponseQuery &
+			PaginationArguments<PaginatedTransactionResponse['nextCursor']> &
+			OrderArguments,
+	): Promise<PaginatedTransactionResponse> {
+		return await this.transport.request({
+			method: 'suix_queryTransactionBlocks',
+			params: [
+				{
+					filter: input.filter,
+					options: input.options,
+				} as SuiTransactionBlockResponseQuery,
+				input.cursor,
+				input.limit,
+				(input.order || 'descending') === 'descending',
+			],
+		});
+	}
+
+	async getTransactionBlock(input: {
+		digest: TransactionDigest;
+		options?: SuiTransactionBlockResponseOptions;
+	}): Promise<SuiTransactionBlockResponse> {
+		if (!isValidTransactionDigest(input.digest)) {
+			throw new Error('Invalid Transaction digest');
+		}
+		return await this.transport.request({
+			method: 'sui_getTransactionBlock',
+			params: [input.digest, input.options],
+		});
+	}
+
+	async multiGetTransactionBlocks(input: {
+		digests: TransactionDigest[];
+		options?: SuiTransactionBlockResponseOptions;
+	}): Promise<SuiTransactionBlockResponse[]> {
+		input.digests.forEach((d) => {
+			if (!isValidTransactionDigest(d)) {
+				throw new Error(`Invalid Transaction digest ${d}`);
+			}
+		});
+
+		const hasDuplicates = input.digests.length !== new Set(input.digests).size;
+		if (hasDuplicates) {
+			throw new Error(`Duplicate digests in batch call ${input.digests}`);
+		}
+
+		return await this.transport.request({
+			method: 'sui_multiGetTransactionBlocks',
+			params: [input.digests, input.options],
+		});
+	}
+
+	async executeTransactionBlock(input: {
```

Comment:
We talked about making this directly take a keypair too, not sure if you were planning on doing that in a followup?

Reply:
Yeah, I think I want to save that for a follow up when we have the keypair changes merged.  I mostly wanted to get the new classes/exports/constructors in place in this PR and make sure everything still works before diverging too far from how the methods themselves work.  Also figured it's easier to review if I do those separately rather than including it in a huge new file